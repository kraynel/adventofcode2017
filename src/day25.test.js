const { executeProgram } = require("./day25");
const fs = require("fs");

describe("executeProgram", () => {
  it("should get the debug value", () => {
    expect(
      executeProgram(`Begin in state A.
Perform a diagnostic checksum after 6 steps.

In state A:
  If the current value is 0:
    - Write the value 1.
    - Move one slot to the right.
    - Continue with state B.
  If the current value is 1:
    - Write the value 0.
    - Move one slot to the left.
    - Continue with state B.

In state B:
  If the current value is 0:
    - Write the value 1.
    - Move one slot to the left.
    - Continue with state A.
  If the current value is 1:
    - Write the value 1.
    - Move one slot to the right.
    - Continue with state A.`)
    ).toBe(3);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day25.txt", "utf8", (err, data) => {
      expect(executeProgram(data)).toMatchSnapshot();
      done();
    });
  });
});
