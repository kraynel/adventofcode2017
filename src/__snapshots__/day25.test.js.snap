// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`executeProgram should get the value 1`] = `5744`;
